AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventPublisher.swift

```diff
@@ -156,12 +156,18 @@ final class IncomingAsyncSubscriptionEventPublisher: Cancellable {
     ) -> GraphQLSubscriptionOperation<Payload> {
 
         let request: GraphQLRequest<Payload>
-        if let auth = auth, modelType.schema.hasAuthenticationRules, let authUser = auth.getCurrentUser() {
-            // TODO: check model schema for identityClaim to figure out which is the ownerId field coming from
-            // https://github.com/aws-amplify/amplify-ios/issues/485
+        if modelType.schema.hasAuthenticationRules,
+            case .success(let claims) = AWSAuthService().getTokenClaims() {
```

Comment:
Instantiating AWSAuthService looks a bit odd here since we're already passing in `AuthCategoryBehavior` into this method. It doesn't look like you can get away with just using the Auth category because once you fetch the session, you'll have to cast it to `AuthCognitoTokensProvider` from AWSPluginsCore and then get the CognitoToken. 

I think we should add the `getTokenClaims` to `AWSAuthServiceBehavior` so you can pass it into this method, and then AWSAuthService can be instantiated in the contructor of IncomingAsyncSubscriptionEventPublisher
```
init(modelType: Model.Type, api: APICategoryGraphQLBehavior, auth: AuthCategoryBehavior?,
   awsAuthService: AWSAuthServiceBehavior?) {

   let awsAuthService = awsAuthService ?? AWSAuthService()
```

Reply:
great idea! thank you!